component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SchemaSettingsCreator.java

```diff
@@ -30,17 +32,30 @@
 
     /**
      * Creates instance and adds schema property
-     * 
+     *
      * @param iNode Node for which parameters are created
      * @param category Parameter category
      * @param redrawParameter Parameter which defines whether UI should be redrawn
      */
     public SchemaSettingsCreator(final IElement iNode, final EComponentCategory category,
             final ElementParameter redrawParameter) {
         super(iNode, category, redrawParameter);
-        if (category == EComponentCategory.BASIC) {
+    }
+
+    public List<ElementParameter> getSettings() {
+        boolean hasSchema = false;
+
+        for (TaCoKitElementParameter param : settings.values()) {
+            if (param.getFieldType() == EParameterFieldType.SCHEMA_TYPE) {
+                hasSchema = true;
+                break;
+            }
+        }
+        if (!hasSchema) {
```

Comment:
please, extract  `createSchemaParameter(String schemaName)` method and move it to parent class to avoid duplicated code